src/app/editor-tabs/ace.component.ts

```diff
@@ -83,6 +88,26 @@ export class AceComponent implements AfterViewInit {
     this.messagingService.publish(events.EDITOR_DIRTY_CHANGED, dirtyState);
   }
 
+  private findSyntaxDefinitionFile(): Promise<string> {
+    return this.syntaxHighlightingService.getSyntaxHighlighting(this.getFileExtension())
+      .then(path => {
+        console.log(`using syntax highlighting rules from "${path}"`);
+        return path;
+      })
+      .catch(() => {
+        console.log(`no syntax highlighting rules available for this file type ("${this.path}")`);
+        return 'none';
+      });
+  }
+
+  private getFileExtension(): string {
+    const extensionBeginsAt = this.path.lastIndexOf('.') + 1;
+    if (extensionBeginsAt !== -1 && this.path.length > extensionBeginsAt) {
```

Comment:
-1 will probably never happen, since 1 is added, maybe 1 should be added after this test